### DISCUSSION TITLE
Editting Bones and Re-exporting them

### DISCUSSION THREAD

**Discussion by HackerAxe** [BadDogSkyrim/PyNifly#237] [Q&A]
Hello! I am making a custom race and I would like to reposition some bones to better fit my race. I made empty parent NiNodes to the nodes I want to move, but upon exporting there was nothing here in the nif. I decided to get a normal skeleton, import it into Blender (there was no issues with the import) then export it out to see if I got the same skeleton back, and nothing appeared.

I am using Blender 3.3. I imported it with these settings (see first picture) then exported them with these settings. (See second picture). I pressed 'A' in Blender to have everything in the scene selected before export.

![image](https://github.com/user-attachments/assets/e1746933-e2e4-41f8-a3ae-c1f04b177f13)
![image](https://github.com/user-attachments/assets/3fa15c5c-c5c7-4d45-8f72-cc8c8ae32aac)

Checking in nifskope, nothing is here. My character is invisible in-game as a result.
![image](https://github.com/user-attachments/assets/4331cc7a-b34b-44da-9caa-72b44e1c530f)

Am I missing something that I should be reading to understand why this is not working?

**Comment by BadDogSkyrim** [comment]
Select Preserve Bone Hierarchy.

If that doesn't work try with the latest version of Blender. I try to keep everything working on older versions, but there's a lot of churn.

**Reply by HackerAxe** [reply]
I took a break from modding but I've come back to use Blender 4.2, and the same issue still occurs that if I import a skeleton (I have the skeleton armature and the collisions selected) then export it, it's a file that's only 1kb big. But if I only select the skeleton armature, I get a skeleton that's 59KB when the original was 74KB.
![image](https://github.com/user-attachments/assets/092494c9-ca63-44a0-b569-acd3ae5c9169)
![image](https://github.com/user-attachments/assets/065e2b2e-6c41-4c61-a336-19f6c62bfc4c)
![image](https://github.com/user-attachments/assets/77259207-e48b-4665-b0c7-584e73926152)

Looking inside, there seems to be some information missing. I'm assuming it's because of me not exporting the bhk capsule shapes?
![image](https://github.com/user-attachments/assets/400d83ef-789f-4e76-b72d-41cc21a58c93)

What would be the proper way to import a skeleton, edit it, then export it back?

**Comment by HackerAxe** [comment]
Disregard my earlier replies, I figured out what to do to get what I needed done, now. My problem was that I shouldn't have imported the collisions and animations if I didn't need to use them. I also shouldn't have renamed the bones at all. Then once I preserve the hierarchy, the skeleton worked just the way it was supposed to. At first I thought it had something to do with the differences I saw in Nifskope, but it didn't seem to actually do anything wrong in game. I also found out I need to add parent nodes to each of the bones I want to move still, but that isn't a problem. I can move the bones in Blender exactly where I want them to without issues.

Thank you for the help!

**Comment by BadDogSkyrim** [comment]
Glad it's working. Importing collisions and animations *shouldn't* have caused problems tho.
